File: code_samples/java/ErrorHandlingRecovery.java
/**
 * Error Handling: Recovery
 * AI/ML Training Sample
 */
public class ErrorHandlingRecovery {
    
    private String data;
    
    public ErrorHandlingRecovery() {
        this.data = "";
    }
    
    public void process(String input) {
        this.data = input;
    }
    
    public String getData() {
        return this.data;
    }
    
    public boolean validate() {
        return this.data != null && !this.data.isEmpty();
    }
    
    public static void main(String[] args) {
        ErrorHandlingRecovery instance = new ErrorHandlingRecovery();
        instance.process("example");
        System.out.println("Data: " + instance.getData());
        System.out.println("Valid: " + instance.validate());
    }
}

File: code_samples/scala/error_handling_recovery.scala
/**
 * Error Handling: Recovery
 * AI/ML Training Sample
 */

class Recovery {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object RecoveryExample {
  def main(args: Array[String]): Unit = {
    val instance = new Recovery()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}
